Source/WebCore/platform/graphics/Path.cpp

```diff
@@ -421,6 +421,11 @@ bool Path::isEmpty() const
     return false;
 }
 
+bool Path::isSingleDataLine() const
+{
+    return !!singleDataLine();
+}
```

Comment:
This will work only if the Path is a PathSegment or a PathStream. It will not work if it's PathCG.

Reply:
Changed it to `definitelySingleLine`.

Reply:
This is not what I meant. What I meant is `singleDataLine() `returns a `PathDataLine` which is a `PathSegment` which can be inside a `PathStream`. But it does not answer the question whether the path is going to draw a single line or not.

Reply:
> This will work only if the Path is a PathSegment or a PathStream.

So, as you said, it works when this is `PathSegment` or `PathStream`, right?
And for `PathCG` for example, `singleDataLine` returns `std::nullopt`. So why doesn't `definitelySingleLine` work? If it returns `true`, then it is definitely a single line.
Or doesn't it work even with `PathSegment` or `PathStream`?